version: "3.9"
services:
  elasticsearch:
    image: docker.elastic.co/elasticsearch/elasticsearch:8.15.0
    container_name: elasticsearch
    environment:
      - discovery.type=single-node
      # Disable xpack security (for development purposes)
      - xpack.security.enabled=false
    ports:
      - "9200:9200"
      - "9300:9300"
    networks:
      - elastic
    volumes:
      - ./elasticsearch/data:/usr/share/elasticsearch/data
    healthcheck:
      test: ["CMD-SHELL", "curl -f http://localhost:9200/_cluster/health || exit 1"]
      interval: 30s
      timeout: 10s
      retries: 5
      start_period: 40s
  app:
    image: python:alpine3.20
    # image: ghcr.io/andrii-rieznik/python-gdal:latest
    container_name: app
    volumes:
      - ./elasticsearch:/app
      - ./elasticsearch/elastic.config.yml:/app/elastic.config.yml
    working_dir: /app
    command: sh -c "pip install -r requirements.txt && python data_loader.py"
    depends_on:
      elasticsearch:
        condition: service_healthy
    networks:
      - elastic
  pygeoapi:
    image: geopython/pygeoapi:latest
    container_name: pygeoapi
    ports:
      - 5000:80
    volumes:
      - ./pygeoapi/pygeoapi.config.yml:/pygeoapi/local.config.yml
    depends_on:
      app:
        condition: service_completed_successfully
    networks:
      - elastic
networks:
  elastic:
    driver: bridge
